### PR TITLE
Add a "prompt_login" helper function

### PR DESCRIPTION
**Pull request by faho**
## Description

This prints a description of the "host". Currently that's

`(chroot:debianchroot) $USER@$hostname`

with the chroot part when needed. That means instead of having one "debian chroot" prompt, we support that out of the box.

This also switches the default and terlar prompts to use it, the other
prompts have slightly different coloring or logic here. It would be possible to allow that by adding options to prompt_host.

Part of #7884.

## TODOs:

- [x] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.rst

## Questions:

- Is this wanted at all? Like I said I think having debian chroots off in a separate prompt just results in nobody using it because they would have to change the prompt *in the chroot* (which means allowing for `fish_config` to work there!), so the alternative seems to be not supporting that at all
- What about the name? `prompt_host` sounds like it might be `prompt_hostname`, but `prompt_chroot_user_at_host` is a bit wordy. `prompt_machine` perhaps?
- Support for things other than the debian chroot thing - do other systems have something similar? How about virtual machines? We already color the host differently if it's ssh.

### CODE REVIEW

**Comment by krobelus** [comment]
(#6398 had a more comprehensive solution, using `systemd-detect-virt` and Debian's `ischroot`, at the cost of adding forks at startup)

Not a strong opinion but I'd be fine with dropping the debian chroot prompt. Or replace it with the generic `systemd-detect-virt --chroot`. (We don't care too much about containers/virtualization, because they hopefully have a different hostname from the host, but the chroot one can be nice.) Though I don't know how often this is useful in practise, is chroot still commonly used (over containers)?.

Distribution integration is a bit lacking in general; I wouldn't be surprised if $PATH was not set up correctly when using fish as login shell on a Debian (chroot) system. That would be nice to have fixed but it's not really within our scope.

`prompt_login` would be an alternative name, if we need it.

**Review by zanchey — APPROVED** [review]

**Comment by faho** [comment]
Alright, renamed to `prompt_login`. It's not perfect, but at least it doesn't have the host/hostname confusion.
